backend/common/src/node_types.rs

```diff
@@ -21,43 +38,70 @@ pub struct NodeDetails {
     pub startup_time: Option<Box<str>>,
 }
 
-#[derive(Deserialize, Debug, Clone, Copy, PartialEq, Eq, Default)]
+///
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
 pub struct NodeStats {
     pub peers: u64,
     pub txcount: u64,
 }
 
+impl Serialize for NodeStats {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
why manual implementation of `serialize` and `deserialize`?

Reply:
Another good candidate to add a comment for I think!

These types are used in feed messages sent to users, and so they came along with custom serialize impls already so that they would serialize to a more compact format. I then added custom deserialize impls so that I could deserialize these things for testing.

But yeah, in summary: we wanted a more compact format than the default serialization would have given :)

Reply:
ok, got it.